apps/balancer-tools/src/app/apr/(utils)/fetchAvgDataForPoolIdDateRange.ts

```diff
@@ -11,13 +12,26 @@ export async function fetchAvgDataForPoolIdDateRange(
   startDate: Date,
   endDate: Date,
 ) {
+  const yieldAprSum = db
+    .select({
+      poolExternalId: yieldTokenApr.poolExternalId,
+      timestamp: yieldTokenApr.timestamp,
+      valueSum: sql<number>`sum(${yieldTokenApr.value})`.as("valueSum"),
+    })
+    .from(yieldTokenApr)
+    .groupBy(yieldTokenApr.poolExternalId, yieldTokenApr.timestamp)
+    .as("yieldTokenAprSum");
   const poolStatsData = await db
     .select({
       poolExternalId: swapFeeApr.poolExternalId,
-      avgApr: sql<number>`cast(avg(   
-        coalesce(${swapFeeApr.value},0) + coalesce(${vebalApr.value},0)
-      ) as decimal)`,
-      avgLiquidity: sql<number>`cast(avg(${poolSnapshots.liquidity}) as decimal)`,
+      avgApr:
```

Comment:
because it does the avg of the values that it finds.

Example: for 4 dates if only one day has veBAL APR and that value is 8, then the avg should be 8/4 = 2, but postgres avg is returning 8, as the other days don't exist on veBAL APR table

Reply:
but why are the other ones missing from the table? 👀 I thought this was the issue we solved originally by populating pool snapshots everyday, even on missing days?

Reply:
yup this for snapshots table, but for apr tables I don't think that makes much sense because we would have a much of rows with 0 APR

Reply:
if it'd make things easier that's ok -- a row costs marginally 0 to us